server/manager/src/main/java/org/apache/accumulo/manager/tableOps/delete/PreDeleteTable.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.manager.tableOps.delete;
+
+import org.apache.accumulo.core.clientImpl.thrift.TableOperation;
+import org.apache.accumulo.core.data.NamespaceId;
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.fate.Repo;
+import org.apache.accumulo.manager.Manager;
+import org.apache.accumulo.manager.tableOps.ManagerRepo;
+import org.apache.accumulo.manager.tableOps.Utils;
+import org.apache.accumulo.manager.tableOps.compact.cancel.CancelCompactions;
+
+public class PreDeleteTable extends ManagerRepo {
+
+  private static final long serialVersionUID = 1L;
+
+  private TableId tableId;
+  private NamespaceId namespaceId;
+
+  public PreDeleteTable(NamespaceId namespaceId, TableId tableId) {
+    this.tableId = tableId;
+    this.namespaceId = namespaceId;
+  }
+
+  @Override
+  public long isReady(long tid, Manager env) throws Exception {
+    return Utils.reserveNamespace(env, namespaceId, tid, false, true, TableOperation.DELETE)
+        + Utils.reserveTable(env, tableId, tid, false, true, TableOperation.DELETE);
+  }
+
+  @Override
+  public Repo<Manager> call(long tid, Manager environment) throws Exception {
+    CancelCompactions.mutateZooKeeper(tid, tableId, environment);
+    return new DeleteTable(namespaceId, tableId);
```

Comment:
I think we still need to unreserve the read lock acquired, even on success, before we enter the DeleteTable repo:

```suggestion
    CancelCompactions.mutateZooKeeper(tid, tableId, environment);
    Utils.unreserveTable(env, tableId, tid, false);
    Utils.unreserveNamespace(env, namespaceId, tid, false);
    return new DeleteTable(namespaceId, tableId);
```

Reply:
Good catch. Working too many things...

Reply:
Fixed in e50c258